test/e2e/tests/test_alert_manager_definition.py

```diff
@@ -0,0 +1,555 @@
+
+        # A successful update could take a little while to complete. 
+        # As a intermediate step, the status should be updated to "UPDATING"
+        # shorly after the update call was made. 
+        time.sleep(UPDATE_WAIT_AFTER_SECONDS)
+        am_resource = k8s.get_resource(am_ref)
+        assert am_resource is not None
+        assert 'status' in am_resource
+        assert 'statusCode' in am_resource['status']
+        assert am_resource['status']['statusCode'] == 'UPDATING'
+
+        assert k8s.wait_on_condition(am_ref, "ACK.ResourceSynced", "True", wait_periods=MAX_WAIT_FOR_SYNCED_MINUTES)
+
+        # After the resource is synced, assert that alert manager is active
+        latest = self.get_alert_manager_definition(prometheusservice_client, workspace_id)
+        assert latest is not None
+        assert 'status' in latest['alertManagerDefinition']
+        assert 'statusCode' in latest['alertManagerDefinition']['status']
+        assert latest['alertManagerDefinition']['status']['statusCode'] == 'ACTIVE'
+        assert 'data' in latest['alertManagerDefinition']
+        assert latest['alertManagerDefinition']['data'].decode('UTF-8') == new_alert_config
+
+        # After updating the resource should be back to active 
+        am_resource = k8s.get_resource(am_ref)
+        assert am_resource is not None
+        assert 'status' in am_resource
+        assert 'statusCode' in am_resource['status']
+        assert am_resource['status']['statusCode'] == 'ACTIVE'
+        condition.assert_synced(am_ref)
+
+
+        # Delete the alert manager definition
+        _, deleted = k8s.delete_custom_resource(am_ref)
+        assert deleted
+
+        # Verify that it is being deleted on the server side
+        latest = self.get_alert_manager_definition(prometheusservice_client, workspace_id)
+        assert 'status' in latest['alertManagerDefinition']
+        assert 'statusCode' in latest['alertManagerDefinition']['status']
+        assert latest['alertManagerDefinition']['status']['statusCode'] == 'DELETING'     
+
+        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+        latest = self.get_alert_manager_definition(prometheusservice_client, workspace_id)
+        assert latest is None
+    
+    def test_failed_alert_manager_creation(self, prometheusservice_client, workspace_resource):
+        # The resource creation can fail 2 ways:
+        #   1) AMP returns an http error right away such as a validationexception
+        #   2) successfull HTTP request, alert manager is "CREATING" then a while after status is "CREATION_FAILED"
+        #       - Can happen for both internal AMP errors and validation errors.  
+
+        # The first error is a regular exception that the controller handles the same for all controllers. 
+        # In this test, we will be testing the 2nd error where the creation doesn't fail right away. 
+
+        sns_topic_name = get_bootstrap_resources().AlertManagerSNSTopic.name_prefix
```

Comment:
Does the `AlertManagerSNSTopic.name_prefix` contain the full topic name?

Reply:
Good catch, should be fixed now.